src/components/Confirm/Confirm.tsx

```diff
@@ -9,42 +10,108 @@ import {
   Anchor,
   Button,
 } from '@keen.io/ui-core';
+import { BodyText } from '@keen.io/typography';
 
-import { Title, Close, Footer, Description, Name } from './Confirm.styles';
+import {
+  Title,
+  Close,
+  Footer,
+  Description,
+  Name,
+  QueryNotUsed,
+  NoDashboardsInfo,
+} from './Confirm.styles';
+import { DashboardsLoader } from './components';
+
+import { DashboardsList } from '../UpdateSavedQueryModal/components';
 
 import {
   getConfirmation,
   hideConfirmation,
   acceptConfirmation,
 } from '../../modules/app';
 import { savedQuerySelectors } from '../../modules/savedQuery';
+import { AppContext } from '../../contexts';
 
-const Confirm: FC<{}> = () => {
+const Confirm = () => {
   const { t } = useTranslation();
   const dispatch = useDispatch();
   const { visible } = useSelector(getConfirmation);
   const { displayName } = useSelector(savedQuerySelectors.getSavedQuery);
+  const [inViewRef, inView] = useInView();
+
+  const { enableDashboardsConnection } = useContext(AppContext);
 
   const closeHandler = () => dispatch(hideConfirmation());
 
+  const isConnectedDashboardsError = useSelector(
+    savedQuerySelectors.getConnectedDashboardsError
+  );
+
+  const isConnectedDashboardsLoading = useSelector(
+    savedQuerySelectors.getConnectedDashboardsLoading
+  );
+
+  const connectedDashboards = useSelector(
+    savedQuerySelectors.getConnectedDashboards
+  );
+
   return (
     <Modal isOpen={visible} onClose={closeHandler}>
       {() => (
         <>
           <ModalHeader onClose={closeHandler}>
             <Title>{t('confirm.delete_query_title')}</Title>
           </ModalHeader>
-          <Description>
-            {t('confirm.delete_message_head')}
-            <Name>{displayName}</Name>
-            {t('confirm.delete_message_tail')}
+          <Description
+            isOverflow={
+              !inView && connectedDashboards && connectedDashboards.length > 0
+            }
+          >
+            {enableDashboardsConnection && isConnectedDashboardsLoading ? (
+              <DashboardsLoader />
+            ) : (
+              <>
+                {t('confirm.delete_message_head')}
+                <Name>{displayName}</Name>
+                {t('confirm.delete_message_tail')}
+                {enableDashboardsConnection && (
+                  <>
+                    {connectedDashboards && connectedDashboards.length > 0 ? (
+                      <div data-testid="connected-dashboards">
+                        <p>{t('confirm.connected_dashboards_info')}</p>
```

Comment:
`<p>` may have some styling assigned in keen-web for example that can break the layout